jina/types/document/multimodal.py

```diff
@@ -44,10 +44,17 @@ def __init__(self, document: Optional[DocumentSourceType] = None,
         super().__init__(document=document, copy=copy, **kwargs)
         if chunks or modality_content_map:
             if chunks:
-                granularities = [chunk.granularity for chunk in chunks]
-                if len(set(granularities)) != 1:
+                g = {c.granularity for c in chunks}
+                if len(g) != 1:
                     raise BadDocType('Each chunk should have the same granularity.')
                 self.chunks.extend(chunks)
+
+                # in case chunks have granularity defined, override
```

Comment:
a warning should be raised if something is overriden for the user?

Reply:
this is your original logic in the code, which I found hard to understand.

Reply:
i keep it as it is as your original does not raise warning

Reply:
>this is your original logic in the code, which I found hard to understand.

to be clear, I'm against the logic that a child granularity can decide parent's granularity